### PR TITLE
ACN to valory + docs

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

1. Changing authorship of ACN connection and protocol. 
2. Updating ACN documentation accordingly.
3. Updating ACN tests accordingly
4. Pinning protobuf ==3.19.4 in Pipfile and tox.ini
5. Updating workflow.yml according to point 4.

## Fixes

- changed authorship on ACN protocol and connection.
- changed `p2p_libp2p*` connection versions to 0.1.0 
- updating documentation accordingly and included p2p-connection.md 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code co- left protocol version at 1.1.0verage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/p2p-connection.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
+The `open_aea/p2p_libp2p:0.26.0` connection allows AEAs to create a peer-to-peer communication network. In particular, the connection creates an overlay network which maps agents' public keys to IP addresses.
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:1]: Care as you changed version on other branch

File: docs/p2p-connection.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  This does not affect the agents ability to interact with the 
+  ethereum ledger and solely necessitates a separate set of 
+  cosmos keys for establishing the libp2p connection.
+</p>
+</div>
+
+First, make sure you have installed the crypto plugin
+of the target test-net. 
+``` bash
+pip install open-aea-ledger-cosmos
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:18]: & open-aea-ledger-ethereum

File: docs/p2p-connection.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+aea config set agent.default_connection open_aea/p2p_libp2p:0.26.0
+aea install
+aea build
+```
+
+Establish the <a href="../por">proof of representation</a>:
+
+``` bash
+aea generate-key cosmos
+aea add-key cosmos cosmos_private_key.txt
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:38]:
```suggestion
aea add-key ethereum ethereum_private_key.txt
```
we want ethereum key skill-side

File: docs/p2p-connection.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+aea add connection open_aea/p2p_libp2p:0.26.0
+aea config set agent.default_connection open_aea/p2p_libp2p:0.26.0
+aea install
+aea build
+```
+
+Establish the <a href="../por">proof of representation</a>:
+
+``` bash
+aea generate-key cosmos
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:37]:
```suggestion
aea generate-key ethereum
```

File: docs/p2p-connection.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+aea add-key cosmos cosmos_private_key.txt
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:70]: see abovde

File: docs/acn-internals.md [hunk 6]
```diff
@@ -578,27 +578,25 @@ similarly for what has been described for the envelope entrance
...
 refer to <a href="../connection/" target="_blank">this guide</a>.
 
-### The `fetchai/p2p_libp2p` connection
+### The `p2p_libp2p` connection
 
-The source code of the `fetchai/p2p_libp2p` connection  
-can be downloaded from 
-<a href="https://aea-registry.fetch.ai/details/connection/fetchai/p2p_libp2p/latest" target="_blank">the AEA Registry</a>,
-or from <a href="https://github.com/fetchai/agents-aea/tree/main/packages/fetchai/connections/p2p_libp2p" target="_blank">the main AEA framework repository.</a>
+The source code of the `p2p_libp2p` connection can be downloaded
```
Comment (DavidMinarsch) [inline docs/acn-internals.md:591]: The author should always be part of the package name

File: docs/p2p-connection.md [hunk 7]
```diff
@@ -0,0 +1,194 @@
+The `open_aea/p2p_libp2p:0.26.0` connection allows AEAs to create a peer-to-peer communication network. In particular, the connection creates an overlay network which maps agents' public keys to IP addresses.
```
Comment (DavidMinarsch) [inline docs/p2p-connection.md:1]:
```suggestion
The `valory/p2p_libp2p:0.26.0` connection allows AEAs to create a peer-to-peer communication network. In particular, the connection creates an overlay network which maps agents' public keys to IP addresses.
```

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Now there's an issue on golang tests. I'll let you fix them.

File: aea/helpers/serializers.py [hunk 8]
```diff
@@ -22,7 +22,10 @@
 import copy
 from typing import Any, Dict, Tuple
 
-from google.protobuf.struct_pb2 import ListValue, Struct
+from google.protobuf.struct_pb2 import (  # pylint: disable=no-name-in-module
+    ListValue,
+    Struct,
+)
```
Comment (Adamantios) [inline aea/helpers/serializers.py:28]: Is `# pylint: disable=no-name-in-module` required after pinning `protobuf` version?

File: .github/workflows/workflow.yml [hunk 9]
```diff
@@ -48,12 +48,12 @@ jobs:
...
       - uses: actions/checkout@master
-      - uses: actions/setup-python@master
+      - uses: actions/setup-python@v3
         with:
           python-version: 3.7
-      - uses: actions/setup-go@master
+      - uses: actions/setup-go@v3
         with:
-          go-version: "^1.14.0"
+          go-version: "1.17.7"
```
Comment (DavidMinarsch) [inline .github/workflows/workflow.yml:56]: We use different versions throughout this file. That's not desirable. I propose we use `>=1.14.0 <=1.17.0` throughout as this makes explicit the ranges for which it works.

Comment (DavidMinarsch) [reply .github/workflows/workflow.yml:56]: Feel free to open an issue for it.

File: aea/helpers/serializers.py (hunk 8 above)
Comment (DavidMinarsch) [reply aea/helpers/serializers.py:28]: No it's not required anymore

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**